### PR TITLE
apache: sync options and add tests

### PR DESCRIPTION
**Pull request by sileht**
apache: sync options and add tests

This change adds test coverage for the apache plugins.

And two missing options SSLCiphers and Timeout.

### CODE REVIEW

File: spec/classes/collectd_plugin_apache_instance_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    SSLCiphers "TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256"
+    Timeout 120
+  </Instance>
+</Plugin>
+EOS
+          is_expected.to contain_file('apache-instance-site2.conf').with(
+            content: content_instance_file,
+            path: "#{options[:plugin_conf_dir]}/25-apache-instance-site2.conf"
+          )
+        end
```
Comment (bastelfreak) [inline spec/classes/collectd_plugin_apache_instance_spec.rb]: could you add another test to verify that it compiles with all dependencies?

File: spec/classes/collectd_plugin_apache_spec.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        context ':manage_package => false on osfamily => RedHat' do
+          let :params do
+            {
+              manage_package: false
+            }
+          end
+
+          it 'Will not manage collectd-apache' do
+            is_expected.not_to contain_package('collectd-apache').with(ensure: 'present',
+                                                                       name: 'collectd-apache')
```
Comment (bastelfreak) [inline spec/classes/collectd_plugin_apache_spec.rb]: same as above

**Comment by bastelfreak** [comment]
Hi @sileht, thanks for the PR. Can you take a look at the inline comments and the failing travis job?

File: spec/classes/collectd_plugin_apache_instance_spec.rb (hunk 1 above)
Comment (sileht) [reply spec/classes/collectd_plugin_apache_instance_spec.rb]: I have added check that required class are included, is that enough ?

Comment (bastelfreak) [reply spec/classes/collectd_plugin_apache_instance_spec.rb]: please add a `is_expected.to compile.with_all_deps`

Comment (sileht) [reply spec/classes/collectd_plugin_apache_instance_spec.rb]: gotcha, done

File: spec/classes/collectd_plugin_apache_spec.rb (hunk 2 above)
Comment (sileht) [reply spec/classes/collectd_plugin_apache_spec.rb]: done

**Pull request merged**